src/lib/Libifl/pbsD_runjob.c

```diff
@@ -122,6 +161,68 @@ __runjob_helper(int c, char *jobid, char *location, char *extend, int req_type)
 	return rc;
 }
 
+/**
+ * @brief	Helper function for pbs_asynrunjob and pbs_asynrunjob_ack
+ *
+ * @param[in] c - connection handle
+ * @param[in] jobid- job identifier
+ * @param[in] location - string of vnodes/resources to be allocated to the job
+ * @param[in] extend - extend string for encoding req
+ * @param[in] req_type - one of PBS_BATCH_AsyrunJob or PBS_BATCH_AsyrunJob_ack
+ *
+ * @return      int
+ * @retval      0       success
+ * @retval      !0      error
+ */
+static int
+__runjob_helper(int c, char *jobid, char *location, char *extend, int req_type)
+{
+	int rc = 0;
+	svr_conn_t **svr_conns = get_conn_svr_instances(c);
+	int i;
+	int start = 0;
+	int ct;
+	int nsvrs = get_num_servers();
+	char *dest = NULL;
+
+	if ((jobid == NULL) || (*jobid == '\0'))
+		return (pbs_errno = PBSE_IVALREQ);
+
+	if (svr_conns) {
```

Comment:
get_conn_svr_instances() will be able to fetch a connection list only if the passed in fd is represents a cluster fd. Otherwise it can be an fd of an instance, hence we fall back and process only that fd instead of looping through.

Reply:
"get_conn_svr_instances() will be able to fetch a connection list only if the passed in fd is represents a cluster fd. "

i know, i was just saying that from the code it's not obvious, so add a comment saying the same thing. I won't block your PR for this, up to you

Reply:
I've added more comments.